diff --git a/src/include/tev.h b/src/include/tev.h
index b7c273e..18a6c01 100644
--- a/src/include/tev.h
+++ b/src/include/tev.h
@@ -20,7 +20,7 @@ typedef enum {
 #define TEV_HANDLE_FIELDS \
     void *data; \
     tev_loop_t *loop; \
-    QUEUE node[2]; \
+    QUEUE queue[2]; \
     tev_handle_type_t type; \
     void (*process)(tev_handle_t *); \
     int is_cancel;
@@ -30,15 +30,19 @@ typedef enum {
     uint64_t repeat; \
     void (*cb)(tev_timer_t *);
 
+#define TEV_HANDLE_IDLE_FILEDS \
+
 
 typedef struct tev_handle_s tev_handle_t;
 typedef struct tev_timer_s tev_timer_t;
 typedef struct tev_io_s tev_io_t;
+typedef struct tev_idle_s tev_idle_t;
 
 
 typedef void (*tev_handle_cb)(tev_handle_t *);
 typedef void (*tev_timer_cb)(tev_timer_t *);
 typedef void (*tev_io_cb)(tev_io_t *);
+typedef void (*tev_idle_cb)(tev_idle_t *);
 
 
 typedef struct {
@@ -68,6 +72,11 @@ struct tev_io_s {
     TEV_HANDLE_FIELDS
 };
 
+struct tev_idle_s {
+    TEV_HANDLE_FIELDS
+    TEV_HANDLE_IDLE_FILEDS
+};
+
 
 /* private APIS */
 int
diff --git a/src/tev_core.c b/src/tev_core.c
index 18f479a..65293d7 100644
--- a/src/tev_core.c
+++ b/src/tev_core.c
@@ -20,7 +20,7 @@ tev__process_handle(tev_loop_t *loop)
     tev_handle_t *handle;
 
     QUEUE_FOREACH(q, loop->handle_queue) {
-        handle = QUEUE_DATA(q, tev_handle_t, node);
+        handle = QUEUE_DATA(q, tev_handle_t, queue);
 
         if (handle->process != NULL) {
             handle->process(handle);
diff --git a/src/tev_timer.c b/src/tev_timer.c
index 5ac7a91..ef2e4da 100644
--- a/src/tev_timer.c
+++ b/src/tev_timer.c
@@ -34,7 +34,7 @@ tev_timer_start(tev_timer_t *handle,
     handle->repeat = repeat;
     handle->cb = cb;
 
-    QUEUE_INSERT_TAIL(handle->loop->handle_queue, handle->node);
+    QUEUE_INSERT_TAIL(handle->loop->handle_queue, handle->queue);
 
     return 0;
 }
@@ -44,7 +44,7 @@ tev_timer_stop(tev_timer_t *handle)
 {
     handle->is_cancel = 1;
 
-    QUEUE_REMOVE(handle->node);
+    QUEUE_REMOVE(handle->queue);
 
     return 0;
 }
